doc/filters/spaceless.rst

```diff
@@ -3,7 +3,8 @@
 
 .. warning::
 
-    The ``spaceless`` filter is deprecated as of Twig 3.12.
+    The ``spaceless`` filter is deprecated as of Twig 3.12. Instead, use the
+    :ref:`whitespace control features <templates-whitespace-control>`.
```

Comment:
```suggestion
    The ``spaceless`` filter is deprecated as of Twig 3.12. If needed, use the
    :ref:`whitespace control features <templates-whitespace-control>`.
```

This is the best suggestion I can come up with which doesn't imply that those controls are a direct replacement without making the message too wordy.